### DISCUSSION TITLE
How to do joints in Radioss?

### DISCUSSION THREAD

**Discussion by Hyp5891** [OpenRadioss/OpenRadioss#2164] [Q&A] [answered]
Hi,

I use ls-prepost to prepare my models. So far I am not able to successfully run a working revolute joint. There are some tutorials or also example files (like this one: https://www.dynaexamples.com/intro-by-j.-day/joint/joint-i which is not working due to "unrecognized option" errors) but until now nothing was working for me. On youtube I found some tutorials but these require software like Hypermesh. To be honest the procedure to create such joints in ls-dyna feels quite cumbersome to me.

Recently I found the "connector tool" in ls prepost which seems to be quite convenient even if it can only create revolute joints...
![image](https://github.com/OpenRadioss/OpenRadioss/assets/144137158/5304f70a-3cfd-4639-8449-91c2e9b1b381)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/144137158/407c2ad9-4da3-46fa-9ae6-9b4b077b8739)

When I run this i got an error in the radioss joint definition that one node does not have a mass. I then added a small mass using *Element Mass. After this, solver is not showing any error but the revolute joint is not working.

![image](https://github.com/OpenRadioss/OpenRadioss/assets/144137158/a350aa44-edd2-457f-a45f-55d8ff7fdadf)

Can somebody please give an instruction and/or example file on how to do this? 

I read this can also be done by defining a spring (TYPE33 (PROP/KJOINT))? For me it would be really satisfying when I could somehow stay inside my ls prepost environment or with a little bit of hand-editing of the .rad-file.

Thanks a lot in advance!
Hyp

**Comment by bertrand-altair** [comment]
Hi @Hyp5891,

I don't know ls-prepost, so i don't know why the model is not working. But i guess it's related to the fact that some features are not yet supported in RADIOSS such as *CONSTRAINED_NODAL_STIFFNESS_GENERALIZED. 

Concerning joints in RADIOSS, they are represented using spring with PROP/TYPE45 (KJOINT2), TYPE33 is considered as obsolete.

Here is small example of a revolute JOINT (in different formats) connecting two rigid bodies. I hope it can help.

[Revolute_joint.zip](https://github.com/OpenRadioss/OpenRadioss/files/13636502/Revolute_joint.zip)

Thank you,

Bertrand,

**Comment by Hyp5891** [comment, marked as answer]
Hey @bertrand-altair, thanks for your feedback!

After trying several things (your example but its far to cumbersome, radioss code in an include file,...) I could come up with a solution of defining a revolute joint that was properly translated into openradioss.

I found it by studying the 1M neo car model. It uses the ls dyna 'MAT196-General_Spring_Discrete_Beam material model which translates into Radioss [/MAT/LAW108](https://2021.help.altair.com/2021/hwsolvers/rad/topics/solvers/rad/mat_law108_spr_gene_starter_r.htm). Depending on what kind of joint is needed, corresponding stiffness values for the 6 DOFs are defined. This material is assigned to a discrete beam element that (in my case) joins two CNRBs.

![image](https://github.com/OpenRadioss/OpenRadioss/assets/144137158/d8359455-86cc-48a4-98bc-eb5b8dc62987)
model

https://github.com/OpenRadioss/OpenRadioss/assets/144137158/958691a7-0523-4ebb-a6da-924f4d8ba901
animation

[2023_12_11_ls_dyna_revolute.k.txt](https://github.com/OpenRadioss/OpenRadioss/files/13650107/2023_12_11_ls_dyna_revolute.k.txt)
ls dyna .k-file

Maybe this is also helpful for somebody else. :)

Best regards,
Hyp

P.S.: I use ls-prepost because there is no other free preprocessor for radioss/ls-dyna that i know of. I would suggest to add it the the Confluence WIKI.

**Reply by PaulAltair** [reply]
I'm not sure this method will be working reliably for revolute joint? a spring with Mat 108 would (should) ordinarily be zero length and dofs are global, so this may be working if your joint is aligned with a global axis, but probably would not if it isn't? Using Mat 113 would be better I think (the axis of the spring is N1-N2 then)

**Reply by PaulAltair** [reply]
actually, probably ok here, as you have systems defined on the section_beam cards that map across

But, your example .k model fails for me with added mass? I have to reduce timestep by factor of 100 (3e-8) to get it running
